lib/data/external/AzureClient.js

```diff
@@ -125,6 +236,74 @@ class AzureClient {
         });
     }
 
+    uploadPart(request, streamingV4Params, partStream, size, key, uploadId,
```

Comment:
As discussed, we have to determine what happens if you have an existing blob called foo and then you put a block on it. If it erases the original block, we have a problem since our metadata will still think it is there.

Reply:
There is a similar issue on delete.  If someone deletes foo and our metadata thinks it is there because there was a prior foo and it actually wipes out our ongoing block list we have a problem.  There are two types of delete with the azure sdk -- delete and deleteifExists. deleteIfExists should not delete your blocks so should be used on a delete call.

Reply:
We are not using deleteIfExists above (around line 100). I think we need to or else we might delete an ongoing mpu.  Also, we need a test where there was an object foo.  then we start an mpu with the name foo.  then there is a delete call to foo.  the original foo object should be deleted but the in progress mpu should not be "harmed" (confirm by completing the mpu and making sure you can get the data.